### ISSUE TITLE
About SDD_nsp_wo.pt

### ISSUE THREAD

**Issue by sunny123123** [realcrane/Human-Trajectory-Prediction-via-Neural-Social-Physics#7]
@JiangbeiYue  Hi, thanks for your great job. But I have a question.
SDD_nsp_wo.pt in saved_models directory，is it the trained weights or  is it just a initial weights? I test a video from SDD dataset with  the code you provided . But  I got a puzzled result.
If I don't load any weights just with random weights, I got the predicted result.
![image](https://user-images.githubusercontent.com/10163641/209626773-7c599d96-e7b5-4390-91e8-13d691504365.png)

if I load SDD_nsp_wo.pt file as NSP weights, I got the similar result.
![image](https://user-images.githubusercontent.com/10163641/209626731-41f9216f-82d3-4e05-b86f-62c696a824f6.png)

so , I mean the predicted result is similar whether loaded or not loaded SDD_nsp_wo.pt.

Thanks for you any answer

**Comment by JiangbeiYue** [comment]
The SDD_nsp_wo.pt is the trained model. First, we can see that these two are similar but different. Here are four possible reasons. The first reason may come from the limited data trials (just one video). The second reason is that our model is based on a physical model and some hyper-parameters regarded as prior knowledge are used. The physical model can ensure that the trajectory ends near the predicted destinations. So, you can see that your two trajectories have very similar destinations. We use prior knowledge and sigmoid functions to restrict the outputs of the neural network to avoid abnormal outputs. Therefore, you can also get not bad results from the untrained model especially when you adopt the best of 20 protocol. The third reason is that the environment parameter we gave is near the best value. The last reason comes from the predicted destinations. I assume that you used the predicted destinations we provided directly. The predicted destinations are predicted by the trained Goal Sampling Network detailed in our supplementary material. If you use all networks including the Goal Sampling Network with initial weights, I think you will get very different results. Hope this will help you.

**Comment by sunny123123** [comment]
@JiangbeiYue 
oh, thanks very much for your reply. I also finded the  goals sampled from Goal Sampling Network are crucial to the final results. 
I also have a question about goals_Ynet.pickle.
```
goals_path = 'data/SDD/goals_Ynet.pickle'
with open(goals_path, 'rb') as f:
    goals = pickle.load(f)
```
goals[1] is the 20 sampled goals from Goal Sampling Network. goals[2] is 16 train video names.  goals[1] shape is [20, pedestrians,2], goals[0] shape is [pedestrians, 2]. **But what is goals[0]? ** it also looks like goals. but I'm not sure. I can't find any explains for this.
Looking forward to your reply again.

**Comment by JiangbeiYue** [comment]
@sunny123123 
I'm sorry that I also forget the accurate meaning of goals[0]. Maybe I just used goals[0] to test something. I think you can just ignore it.

**Comment by sunny123123** [comment]
ok， thanks for you reply again. I will close this issue.
